backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -66,9 +67,56 @@ public UserInfo createUser(
   public UserInfo createUserInCurrentOrg(
       String username, PersonName name, Role role, boolean active) {
     Organization org = _orgService.getCurrentOrganization();
+
+    Optional<ApiUser> found = _apiUserRepo.findByLoginEmailIncludeArchived(username);
+    if (found.isPresent()) {
+      return reprovisionUserHelper(found.get(), name, org, role);
+    }
+
     return createUserHelper(username, name, org, role, true);
   }
 
+  private UserInfo reprovisionUserHelper(
+      ApiUser apiUser, PersonName name, Organization org, Role role) {
+    if (!apiUser.isDeleted()) {
+      // an enabled user with this email address exists (in some org)
+      throw new ConflictingUserException("A user with this email address already exists.");
+    }
+
+    OrganizationRoleClaims claims =
+        _oktaRepo
+            .getOrganizationRoleClaimsForUser(apiUser.getLoginEmail())
+            .orElseThrow(MisconfiguredUserException::new);
+    if (!org.getExternalId().equals(claims.getOrganizationExternalId())) {
+      throw new ConflictingUserException(
+          "The user already exists, but is not in your organization.");
+    }
+
+    // re-provision the user
+    apiUser.setNameInfo(name);
```

Comment:
perform the DB operations after the Okta operations, since Okta has no rollback mechanism currently, so if an Okta calls fails after a DB call, you end up in an inconsistent state.